Tests/SonarScanner.MSBuild.PreProcessor.Test/CacheProcessorTests.cs

```diff
@@ -0,0 +1,53 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarScanner.MSBuild.Common;
+
+namespace SonarScanner.MSBuild.PreProcessor.Test
+{
+    [TestClass]
+    public class CacheProcessorTests
+    {
+        public TestContext TestContext { get; set; }
+
+        [TestMethod]
+        public void Constructor_NullArguments_Throws()
+        {
+            var server = new Mock<ISonarQubeServer>().Object;
+            var settings = CreateProcessedArgs();
+            var logger = new Mock<ILogger>().Object;
+            ((Func<CacheProcessor>)(() => new CacheProcessor(server, settings, logger))).Should().NotThrow();
+            ((Func<CacheProcessor>)(() => new CacheProcessor(null, settings, logger))).Should().Throw<ArgumentNullException>().Which.ParamName.Should().Be("server");
+            ((Func<CacheProcessor>)(() => new CacheProcessor(server, null, logger))).Should().Throw<ArgumentNullException>().Which.ParamName.Should().Be("settings");
+            ((Func<CacheProcessor>)(() => new CacheProcessor(server, settings, null))).Should().Throw<ArgumentNullException>().Which.ParamName.Should().Be("logger");
```

Comment:
We have multiple casts, duplication and code here is rather hard to read. What if we extract:

```
        private static Func<CacheProcessor> CreateProcessorDelegate(ISonarQubeServer server, ProcessedArgs settings, ILogger logger) =>
            () => new CacheProcessor(server, settings, logger);
```

and then use it like:

```
CreateProcessorDelegate(server, settings, logger).Should().NotThrow();
...
```

Reply:
I tried and it doesn't read better to me. The real problem is the variable length of arguments that makes the `Should()` part misaligned 😒 
```
            CreateDelegate(server, settings, logger).Should().NotThrow();
            CreateDelegate(null, settings, logger).Should().Throw<ArgumentNullException>().Which.ParamName.Should().Be("server");
            CreateDelegate(server, null, logger).Should().Throw<ArgumentNullException>().Which.ParamName.Should().Be("settings");
            CreateDelegate(server, settings, null).Should().Throw<ArgumentNullException>().Which.ParamName.Should().Be("logger");

            Func<CacheProcessor> CreateDelegate(ISonarQubeServer server, ProcessedArgs settings, ILogger logger) =>
                () => new CacheProcessor(server, settings, logger);
```
What about (not so nice, but serves the purpose) `serv`, `sett` and `logg` names instead? To have equal length as `null`

Reply:
But they don't have to be the same length. We don't do ASCII art :)))

Reply:
I mean you can rename them if you really want. I care more about the duplication